### PR TITLE
doc: Add developer notes about gitignore

### PR DESCRIPTION
**Pull request by laanwj**
- Add developer notes about `.gitignore.`
- Remove qt creator specific files from gitignore, to be consistent

Context: discussion in #6813, this is a bit of a FAQ so worth mentioning.

### CODE REVIEW

**Comment by btcdrak** [comment]
Good.

You should mention the simple command line however which does most of the hard work as per https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

**Comment by dexX7** [comment]
Thanks for the information, very helpful!

**Comment by fanquake** [comment]
ACK

**Comment by jonasschnelli** [comment]
ACK.

**Comment by laanwj** [comment]
> You should mention the simple command line however which does most of the hard work as per https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

Added a mention of that - personally I prefer editing the config file manually, instead of memorizing the command, don't think it's very hard work :)

File: doc/developer-notes.md [hunk 1]
```diff
@@ -171,3 +171,36 @@ Threads
 - BitcoinMiner : Generates bitcoins (if wallet is enabled).
 
 - Shutdown : Does an orderly shutdown of everything.
+
+Ignoring IDE/editor files
+--------------------------
+
+In closed-source environments in which everyone uses the same IDE it is common
+to add temporary files it produces to the project-wide `.gitignore` file.
```
Comment (luke-jr) [inline doc/developer-notes.md]: Also fairly standard practice in open-source environments... (Not saying we have to do it, but this description isn't correct.)

Comment (laanwj) [inline doc/developer-notes.md]: In my experience in - much more heterogeneous, multiplatform - open source projects it's rare to see ignores for specific tools that some of the developers use

Comment (btcdrak) [inline doc/developer-notes.md]: I think the point is that ignore files should always be specific to the project, and not to the developer's environment.

Comment (luke-jr) [inline doc/developer-notes.md]: I'm thinking things like *~

**Comment by petertodd** [comment]
ACK

**Comment by MarcoFalke** [comment]
ACK `.gitignore` diff

Meh, "doesn't hurt" on the dev notes.

**Pull request merged**